src/features/activities/ActivityRow.test.tsx

```diff
@@ -0,0 +1,230 @@
+    reactions: [],
+    id: 129,
+    action: 'created',
+    created_at: new Date('2022-03-15T09:30:00'),
+  };
+
+  const Harness = () => {
+    return <ActivityRow activity={activity} />;
+  };
+
+  await act(async () => {
+    await render(
+      <TestWrapper withWeb3>
+        <Harness />
+      </TestWrapper>
+    );
+  });
+
+  expect(screen.getByText(/Unknown activity:/));
+});
+
+test('shows epoch activity row', async () => {
+  const activity: Activity = {
+    id: 114723,
+    action: 'epochs_insert',
+    created_at: new Date().toISOString(),
+    circle: {
+      id: 242,
+      name: 'Tools',
+    },
+    epoch: {
+      start_date: new Date().toISOString(),
+      end_date: new Date(Date.now() + 3 * 24 * 60 * 60 * 1000).toISOString(),
+      number: 1,
+      ended: false,
+    },
+    reactions: [],
+  };
+
+  const Harness = () => {
+    return <ActivityRow activity={activity} />;
+  };
+
+  await act(async () => {
+    await render(
+      <TestWrapper withWeb3>
+        <Harness />
+      </TestWrapper>
+    );
+  });
+
+  expect(screen.getByText('Epoch Created'));
+  expect(screen.getByText('Tools'));
+  expect(screen.getByText(/Starts/));
+});
+
+test('shows new user activity row', async () => {
+  const activity: Activity = {
+    id: 114801,
+    action: 'users_insert',
+    created_at: new Date(Date.now() - 3 * 24 * 60 * 60 * 1000).toISOString(),
+    circle: {
+      id: 242,
+      name: 'Tools',
+    },
+    target_profile: {
+      name: 'Leon',
+      avatar: '🦒',
+      address: '0x14dc79964da2c08b23698b3d3cc7ca32193d9955',
+    },
+    reactions: [],
+  };
+
+  const Harness = () => {
+    return <ActivityRow activity={activity} />;
+  };
+
+  await act(async () => {
+    await render(
+      <TestWrapper withWeb3>
+        <Harness />
+      </TestWrapper>
+    );
+  });
+
+  expect(screen.getByText('Leon'));
+  expect(screen.getByText('joined'));
+  expect(screen.getByText('Tools'));
+  expect(screen.getByText(/days ago/));
+});
+
+test('shows contribution activity row', async () => {
+  const activity: Activity = {
+    id: 114715,
+    action: 'contributions_insert',
+    created_at: new Date(Date.now() - 3 * 24 * 60 * 60 * 1000).toISOString(),
+    actor_profile: {
+      name: 'Francine',
+      avatar: 'https://coordinape-prod.s3.amazonaws.com/sdao-francine.jpg',
+      address: '0xff0ca13819312375171fecd98460e5fb71c7ff6f',
+    },
+    circle: {
+      id: 251,
+      name: 'Tools',
+      logo: 'b90806b4-d724-40c4-b9c1-aa89225844a8.jpg',
+    },
+    contribution: {
+      description: '*Building the integrations page*',
+    },
+    reactions: [],
+  };
+
+  const Harness = () => {
+    return <ActivityRow activity={activity} />;
+  };
+
+  await act(async () => {
+    await render(
+      <TestWrapper withWeb3>
+        <Harness />
+      </TestWrapper>
+    );
+  });
+
+  expect(screen.getByText('Francine'));
+  expect(screen.getByText(/days ago/));
+  // TODO: test the markdown preview
```

Comment:
@levity any idea about this error? The MarkdownPreview component doesn't appear on the page at the time this expect happens... not sure how to fix